src/relax/op/tensor/binary.cc

```diff
@@ -83,12 +92,17 @@ StructInfo InferStructInfoBroadcast(const Call& call, const BlockBuilder& ctx) {
     for (; i <= std::min(lhs_ndim, rhs_ndim); ++i) {
       const PrimExpr& dim0 = lhs_shape->values[lhs_ndim - i];
       const PrimExpr& dim1 = rhs_shape->values[rhs_ndim - i];
-      if (EqualConstInt(dim0, 1)) {
+      if (IsConstInt(dim0, 1)) {
```

Comment:
How about we do
```
const auto* int_dim0 = dim0.as<IntImmNode>();
const auto* int_dim1 = dim1.as<IntImmNode>();
```
Then followup can all become conditions related to constant checks.

```
if (int_dim1 && int_dim1->value == 1)
```

Reply:
do a quick check for int values as well.